sound/soc/sof/intel/hda-ipc.c

```diff
@@ -268,10 +268,13 @@ void hda_ipc_msg_data(struct snd_sof_dev *sdev,
 					  hda_stream.hstream);
 
 		/* The stream might already be closed */
-		if (hstream)
-			sof_mailbox_read(sdev, hda_stream->stream.posn_offset,
-					 p, sz);
+		if (!hstream)
+			return -ENODEV;
```

Comment:
Would -EINVAL or -EIO be a better error code to return? The device is there, but the stream is no longer runs.

Reply:
Or perhaps -EPIPE / -ESTRPIPE ?

Reply:
@ujfalusi don't think `EIO` would be a good choice. I was thinking about `EINVAL` but among other reasons `EINVAL` seems to be (ab)used for nearly all errors... In this case I thought it was like "we didn't find this instance." But we can make it `EINVAL` if more people think it's more appropriate

Reply:
@lyakh, I agree that `EINVAL` is used too widely, but `ENODEV` here does not sound reasonably.

Reply:
I'd vote for ESTRPIPE (but can live with all options)